pkg/collector/corechecks/ebpf/oom_kill.go

```diff
@@ -0,0 +1,153 @@
+
+// FIXME: we require the `cgo` build tag because of this dep relationship:
+// github.com/DataDog/datadog-agent/pkg/process/net depends on `github.com/DataDog/agent-payload/process`,
+// which has a hard dependency on `github.com/DataDog/zstd`, which requires CGO.
+// Should be removed once `github.com/DataDog/agent-payload/process` can be imported with CGO disabled.
+// +build cgo
+// +build linux
+
+package ebpf
+
+import (
+	"fmt"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	dd_config "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/ebpf/oomkill"
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	process_net "github.com/DataDog/datadog-agent/pkg/process/net"
+	"github.com/DataDog/datadog-agent/pkg/tagger"
+	"github.com/DataDog/datadog-agent/pkg/tagger/collectors"
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	oomKillCheckName = "oom_kill"
+)
+
+// OOMKillConfig is the config of the OOMKill check
+type OOMKillConfig struct {
+	CollectOOMKill bool `yaml:"collect_oom_kill"`
+}
+
+// OOMKillCheck grabs OOM Kill metrics
+type OOMKillCheck struct {
+	core.CheckBase
+	instance *OOMKillConfig
+}
+
+// OOMKillFactory is exported for integration testing
+func OOMKillFactory() check.Check {
+	return &OOMKillCheck{
+		CheckBase: core.NewCheckBase(oomKillCheckName),
+		instance:  &OOMKillConfig{},
+	}
+}
+
+func init() {
+	core.RegisterCheck(oomKillCheckName, OOMKillFactory)
+}
+
+// Parse parses the check configuration
+func (c *OOMKillConfig) Parse(data []byte) error {
+	// default values
+	c.CollectOOMKill = true
+
+	if err := yaml.Unmarshal(data, c); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Configure parses the check configuration and init the check
+func (m *OOMKillCheck) Configure(config, initConfig integration.Data, source string) error {
+	// TODO: Remove that hard-code and put it somewhere else
+	process_net.SetSystemProbePath(dd_config.Datadog.GetString("system_probe_config.sysprobe_socket"))
+
+	err := m.CommonConfigure(config, source)
+	if err != nil {
+		return err
+	}
+
+	return m.instance.Parse(config)
+}
+
+// Run executes the check
+func (m *OOMKillCheck) Run() error {
+	if !m.instance.CollectOOMKill {
+		return nil
+	}
+
+	sysProbeUtil, err := process_net.GetRemoteSystemProbeUtil()
+	if err != nil {
+		return err
+	}
+
+	data, err := sysProbeUtil.GetCheck("oom_kill")
+	if err != nil {
+		return err
+	}
+
+	// sender is just what is used to submit the data
+	sender, err := aggregator.GetSender(m.ID())
+	if err != nil {
+		return err
+	}
+
+	triggerType := ""
+	triggerTypeText := ""
+	mainText := ""
+	for _, lineRaw := range data {
+		line := lineRaw.(oomkill.Stats)
```

Comment:
Should we check for the type conversion here? This could panic if `lineRaw` isn't of type `oomkilll.Stats`